utilities/rocAL/rocal_external_source/rocal_external_source.cpp

```diff
@@ -0,0 +1,382 @@
+    labels.resize(inputBatchSize);
+    int iter_cnt = 0;
+    float  pmul = 2.0f/255;
+    float  padd = -1.0f;
+    int index = 0;
+    while (!rocalIsEmpty(handle)) {
+        std::vector<std::string> input_images;
+        std::vector<unsigned char*> input_batch_buffer;
+        std::vector<unsigned> roi_width;
+        std::vector<unsigned> roi_height;
+        std::vector<int> label;
+        for(int i = 0; i < inputBatchSize; i++) {
+            if(mode == 0) {
+                input_images.push_back(file_names.back());
+                file_names.pop_back();
+                if((file_names.size()) == 0) {
+                    eos = true;
+                }
+            } else {
+                if(mode == 1) {
+                    input_batch_buffer.push_back(input_buffer.back());
+                    input_buffer.pop_back();
+                    roi_height.push_back(srcsize_height.back());
+                    srcsize_height.pop_back();
+                } else {
+                    input_batch_buffer.push_back(input_buffer.back());
+                    input_buffer.pop_back();
+                    roi_width.push_back(srcsize_width.back());
+                    srcsize_width.pop_back();
+                    roi_height.push_back(srcsize_height.back());
+                    srcsize_height.pop_back();
+                }
+                if((file_names.size()) == 0 || input_buffer.size() == 0) {
+                    eos = true;
+                }
+            }
+        }
+        if(index <= (total_images / inputBatchSize))
+        {
+            if(mode == 0)
+                rocalExternalSourceFeedInput(handle, input_images, label, {}, {}, {}, decode_width, decode_height, channels, RocalExtSourceMode (0), RocalTensorLayout (0), eos);
+            else if(mode == 1)
+                rocalExternalSourceFeedInput(handle, {}, label, input_batch_buffer, {}, roi_height, decode_width, decode_height, channels, RocalExtSourceMode (mode), RocalTensorLayout (0), eos);
+            else if(mode == 2)
+                rocalExternalSourceFeedInput(handle, {}, label, input_batch_buffer, roi_width, roi_height, maxwidth, maxheight, channels, RocalExtSourceMode (mode), RocalTensorLayout (0), eos);
+        }
+        if(rocalRun(handle) != 0)
+            break;
+
+        if(display)
+            rocalCopyToOutput(handle, mat_input.data, h*w*p);
+        counter += inputBatchSize;
+        iter_cnt++;
+
+        if(!display)
+            continue;
+        std::vector<int> compression_params;
+        compression_params.push_back(IMWRITE_PNG_COMPRESSION);
+        compression_params.push_back(9);
+        std::string out_filename = std::string("output") + std::to_string(index) + ".png";
+        if(color_format ==  RocalImageColor::ROCAL_COLOR_RGB24 ) {
+            mat_input.copyTo(mat_output(cv::Rect(  col_counter*w, 0, w, h)));
+            cv::cvtColor(mat_output, mat_color, CV_RGB2BGR);
+            // cv::imshow("output",mat_color);
```

Comment:
Please remove commented line
Applicable to all such instances in the file

Reply:
Also, adjust the formatting in 
 mat_input.copyTo(mat_output(cv::Rect(  col_counter*w, 0, w, h)));

Reply:
Formated the document